swagger.yaml

```diff
@@ -4294,7 +4294,7 @@ definitions:
         type: string
         description: "The desired network type used."
         enum: ['OpenShiftSDN', 'OVNKubernetes']
-        default: 'OpenShiftSDN'
+        x-nullable: true
```

Comment:
Does this break clients/API?

Reply:
I think it shouldn't break
At the schema-level, we don't disallow anything that wasn't possible before
At the data & values level, UI team is preparing to remove their extra-logic of setting ovn on "ocp>=4.12 || SNO", and provide no value for the network-type (letting us decide on cluster creation what should be the smart-default)

Reply:
I ran through the code and I also think this is a safe change. If user provided the value, we will just carry it. If user does not provide a value, then immediately when we go to cluster creation we will fill that value with something real.

So in the current form it looks like there is no path to end up with empty network type in cluster creation.

How about cluster update though? I may not read it carefully enough but I have a feeling that it’s now possible to unset network type

Reply:
I don't think it is possible to unset the network type in cluster update, because: 
https://github.com/openshift/assisted-service/blob/2ee5cb8ca37b7cf6d96ec27f4d7980aaeea111e6/internal/bminventory/inventory.go#L2350-L2351